tsdb/raw.go

```diff
@@ -93,6 +93,7 @@ func (e *RawExecutor) execute(out chan *models.Row) {
 	// Keep looping until all mappers drained.
 	var err error
 	for {
+		timeout := time.After(30 * time.Second)
```

Comment:
Isn't it possible that a query would not read from the cursor for more than 30s? For example, the query hits 2 shards and the first shard takes more than 30s to process.